google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/BigQueryReadClientTest.java

```diff
@@ -0,0 +1,165 @@
+import io.grpc.StatusRuntimeException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BigQueryReadClientTest {
+  private static MockBigQueryRead mockBigQueryRead;
+  private static MockServiceHelper serviceHelper;
+  private BigQueryReadClient client;
+  private LocalChannelProvider channelProvider;
+
+  @BeforeClass
+  public static void startStaticServer() {
+    mockBigQueryRead = new MockBigQueryRead();
+    serviceHelper =
+        new MockServiceHelper(
+            UUID.randomUUID().toString(), Arrays.<MockGrpcService>asList(mockBigQueryRead));
+    serviceHelper.start();
+  }
+
+  @AfterClass
+  public static void stopServer() {
+    serviceHelper.stop();
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    serviceHelper.reset();
+    channelProvider = serviceHelper.createChannelProvider();
+    BigQueryReadSettings settings =
+        BigQueryReadSettings.newBuilder()
+            .setTransportChannelProvider(channelProvider)
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .build();
+    client = BigQueryReadClient.create(settings);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    client.close();
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void createReadSessionTest() {
+    String name = "name3373707";
+    String table = "table110115790";
+    ReadSession expectedResponse = ReadSession.newBuilder().setName(name).setTable(table).build();
+    mockBigQueryRead.addResponse(expectedResponse);
+
+    String parent = "parent-995424086";
+    ReadSession readSession = ReadSession.newBuilder().build();
+    int maxStreamCount = 940837515;
+
+    ReadSession actualResponse = client.createReadSession(parent, readSession, maxStreamCount);
+    Assert.assertEquals(expectedResponse, actualResponse);
+
+    List<AbstractMessage> actualRequests = mockBigQueryRead.getRequests();
+    Assert.assertEquals(1, actualRequests.size());
+    CreateReadSessionRequest actualRequest = (CreateReadSessionRequest) actualRequests.get(0);
+
+    Assert.assertEquals(parent, actualRequest.getParent());
+    Assert.assertEquals(readSession, actualRequest.getReadSession());
+    Assert.assertEquals(maxStreamCount, actualRequest.getMaxStreamCount());
+    Assert.assertTrue(
+        channelProvider.isHeaderSent(
+            ApiClientHeaderProvider.getDefaultApiClientHeaderKey(),
+            GaxGrpcProperties.getDefaultApiClientHeaderPattern()));
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void createReadSessionExceptionTest() throws Exception {
+    StatusRuntimeException exception = new StatusRuntimeException(Status.INVALID_ARGUMENT);
+    mockBigQueryRead.addException(exception);
+
+    try {
```

Comment:
are we not using junit5 yet?  Otherwise you should be able to use assert exception with a lambda

Reply:
or is supposed to be compatible with java 7?

Reply:
Based on the repo doc, it is compatible with Java 7: https://github.com/googleapis/java-bigquerystorage#java-versions

The changes in this file match the auto-generated unit tests in `BaseBigQueryReadClientTest.java`.